tests/periph_timer/tests/01-run.py

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2016 Kaspar Schleiser <kaspar@schleiser.de>
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+import os
+import sys
+
+sys.path.append(os.path.join(os.environ['RIOTBASE'], 'dist/tools/testrunner'))
+import testrunner
+
+
+def testfunc(child):
+    child.expect('Available timers: \d')
+    timers_num = int(child.after[len('Available timers: '):])
```

Comment:
How about using the features provided by `pexpect` ;-)

```python
child.expect('Available timers: (\d+)')
timers_num = int(child.match.group(1))
```

(also: note the `+` after `\d` for the [unlikely] case a board has more than 9 timers ;-))

Reply:
Thanks for the hint ! I was looking for something like this in the doc but missed it.

Reply:
Read more on `spawn::match` [here](http://pexpect.readthedocs.io/en/stable/api/pexpect.html?highlight=match#pexpect.spawn.expect).

Reply:
Pushed an improved version. Now it supports the unlikely case when a board has more than 9 timers